core/src/main/java/org/apache/calcite/sql/type/ReturnTypes.java

```diff
@@ -620,6 +620,13 @@ public static SqlCall stripSeparator(SqlCall call) {
   public static final SqlReturnTypeInference MULTISET_ELEMENT_NULLABLE =
```

Comment:
Since only the ELEMENT operator is used, I think it can be removed

Reply:
Since it is public I was assuming it may be used in other third-party libraries. What is the policy about removing public fields in Calcite?

Reply:
This is a minor suggestion that can be ignored.
I'm not sure about the removal policy either.  Not deleting can avoid version upgrade issues, but maintenance may be lengthy.

Reply:
My two cents: in this particular case, I'd vote for removing the (now unused) `MULTISET_ELEMENT_NULLABLE`.
It is true that it might be used by a 3rd party library, but since the name was misleading, I think that those potential 3rd party libraries could be suffering from the same issue as CALCITE-6228, so by having a compilation problem after the next Calcite upgrade, they'd be force to analyze their usage of `MULTISET_ELEMENT_NULLABLE` and maybe switch to the new `MULTISET_ELEMENT_FORCE_NULLABLE` (or define their own `SqlReturnTypeInference` if they choose so).

Reply:
I love removing code, I will do that right away.
I will squash the commits optimistically, assuming that it can be merged after that.